Rename `sorted` attribute of `Section` to `is_sorted` to solve conflict with `Section.sorted` static method

Mypy seems to give one type-checking error (on the master branch).

```
cloup/_sections.py:68: error: Function "Callable[[str, Union[Iterable[Command], Dict[str, Command]]], Section]" could always be true in boolean context  [truthy-function]
```

This evidently arises because `Section` has both an instance variable named `sorted` (a bool) and a class method named `sorted` too. I very much think it would be worth renaming one of these variables, since it's only going to create confusion and type-checking complications. (You're already having to use `type: ignore` marker in the constructor.)

--- a/cloup/_commands.py
+++ b/cloup/_commands.py
@@ -527,8 +527,8 @@ def command(
                 raise TypeError(
                     f"a Command must inherit from cloup.ConstraintMixin to support "
                     f"constraints; {cls} doesn't")
-            constraints = tuple(reversed(f.__cloup_constraints__))  # type: ignore
-            del f.__cloup_constraints__  # type: ignore
+            constraints = tuple(reversed(f.__cloup_constraints__))
+            del f.__cloup_constraints__
             kwargs['constraints'] = constraints
 
         cmd_cls = cls if cls is not None else Command

--- a/cloup/_sections.py
+++ b/cloup/_sections.py
@@ -27,18 +27,18 @@ class Section:
 
     def __init__(self, title: str,
                  commands: Subcommands = (),
-                 sorted: bool = False):  # noqa
+                 is_sorted: bool = False):  # noqa
         """
         :param title:
         :param commands: sequence of commands or dictionary {name: command}
-        :param sorted:
+        :param is_sorted:
             if True, ``list_commands()`` returns the commands in lexicographic order
         """
         if not isinstance(title, str):
             raise TypeError(
                 'the first argument must be a string, the title. You probably forgot it.')
         self.title = title
-        self.sorted = sorted  # type: ignore
+        self.is_sorted = is_sorted
         self.commands: OrderedDict[str, click.Command] = OrderedDict()
         if isinstance(commands, Sequence):
             self.commands = OrderedDict()
@@ -52,7 +52,7 @@ class Section:
 
     @classmethod
     def sorted(cls, title: str, commands: Subcommands = ()) -> 'Section':
-        return cls(title, commands, sorted=True)
+        return cls(title, commands, is_sorted=True)
 
     def add_command(self, cmd: click.Command, name: Optional[str] = None) -> None:
         name = name or cmd.name
@@ -65,7 +65,7 @@ class Section:
     def list_commands(self) -> List[Tuple[str, click.Command]]:
         command_list = [(name, cmd) for name, cmd in self.commands.items()
                         if not cmd.hidden]
-        if self.sorted:
+        if self.is_sorted:
             command_list.sort()
         return command_list
 
@@ -73,7 +73,7 @@ class Section:
         return len(self.commands)
 
     def __repr__(self) -> str:
-        return 'Section({}, sorted={})'.format(self.title, self.sorted)
+        return 'Section({}, is_sorted={})'.format(self.title, self.is_sorted)
 
 
 class SectionMixin:

--- a/tests/example_group.py
+++ b/tests/example_group.py
@@ -43,7 +43,7 @@ def make_example_group(align_sections):
     git.add_section(Section(
         'Work on the current change (see also: git help everyday)',
         [git_rm, git_sparse_checkout, git_mv],
-        sorted=True
+        is_sorted=True
     ))
 
     git.add_command(cloup.command('fake-3', hidden=True)(f))

